library/bignum.c

```diff
@@ -512,6 +512,12 @@ int mbedtls_mpi_read_string( mbedtls_mpi *X, int radix, const char *s )
 
     mbedtls_mpi_init( &T );
 
+    if( s[0] == 0 )
+    {
+        mbedtls_mpi_free( X );
+        return( 0 );
```

Comment:
Pre-existing: I don't think it is a good idea to accept the empty string as a valid representation of 0. I was wondering if we could change this on the development branches?

Reply:
I disagree. It's natural that a string of 0 digits represents the value 0. Forcing the presence of at least one digit means that you have to think about this exception whenever you write a number. It also means that the behavior would deviate from `read_binary`, where a 0-length bitstring has to be a valid representation of 0.

Reply:
In both cases I think it is more robust to have the empty string as an error case as that can easily happen inadvertently and are not valid representations of 0 in the strict sense. But of course this is subjective and pre-existing anyway.